src/System.Private.CoreLib/shared/System/Text/EncodingProvider.cs

```diff
@@ -11,29 +12,29 @@ namespace System.Text
     public abstract class EncodingProvider
     {
         public EncodingProvider() { }
-        public abstract Encoding GetEncoding(string name);
-        public abstract Encoding GetEncoding(int codepage);
+        public abstract Encoding? GetEncoding(string name);
+        public abstract Encoding? GetEncoding(int codepage);
 
         // GetEncoding should return either valid encoding or null. shouldn't throw any exception except on null name
-        public virtual Encoding GetEncoding(string name, EncoderFallback encoderFallback, DecoderFallback decoderFallback)
+        public virtual Encoding? GetEncoding(string name, EncoderFallback encoderFallback, DecoderFallback decoderFallback)
```

Comment:
Aren't `encoderFallback` and `decoderFallback` allowed to be null in some cases?

Reply:
Yes it is allowed to be null.

---
In reply to: [272224716](https://github.com/dotnet/coreclr/pull/23708#discussion_r272224716) [](ancestors = 272224716)

Reply:
We do allow not assigning them at all but not setting them explicitly to null. In here this will throw:

```csharp
        public EncoderFallback EncoderFallback
        {
            get
            {
                return encoderFallback;
            }

            set
            {
                if (this.IsReadOnly)
                    throw new InvalidOperationException(SR.InvalidOperation_ReadOnly);

                if (value == null)
                    throw new ArgumentNullException(nameof(value));

                encoderFallback = value;
            }
        }
```